tzif.zig

```diff
@@ -1093,49 +1129,365 @@ test "parse Pacific/Honolulu zoneinfo and calculate local times" {
+    try testing.expectEqual(stdoff, result.offset(1583661599).offset);
+    try testing.expectEqual(dstoff, result.offset(1583661600).offset);
+    // transtion dst 2020-11-01T01:59:59-07:00 --> std 2020-11-01T01:00:00-08:00
+    try testing.expectEqual(dstoff, result.offset(1604221199).offset);
+    try testing.expectEqual(stdoff, result.offset(1604221200).offset);
+}
+
+test "posix TZ string, leap year, America/Sitka" {
+    // IANA identifier: America/Sitka
+    const result = try parsePosixTZ("AKST9AKDT,M3.2.0,M11.1.0");
+    const stdoff: i32 = -32400;
+    const dstoff: i32 = -28800;
+    try testing.expectEqualSlices(u8, "AKST", result.std_designation);
+    try testing.expectEqual(stdoff, result.std_offset);
+    try testing.expectEqualSlices(u8, "AKDT", result.dst_designation.?);
+    try testing.expectEqual(dstoff, result.dst_offset);
+    // transition std 2020-03-08T01:59:59-09:00 --> dst 2020-03-08T03:00:00-08:00
+    try testing.expectEqual(stdoff, result.offset(1583665199).offset);
+    try testing.expectEqual(dstoff, result.offset(1583665200).offset);
+    // transtion dst 2020-11-01T01:59:59-08:00 --> std 2020-11-01T01:00:00-09:00
+    try testing.expectEqual(dstoff, result.offset(1604224799).offset);
+    try testing.expectEqual(stdoff, result.offset(1604224800).offset);
+}
+
+test "posix TZ string, leap year, Asia/Jerusalem" {
+    // IANA identifier: Asia/Jerusalem
+    const result = try parsePosixTZ("IST-2IDT,M3.4.4/26,M10.5.0");
+    const stdoff: i32 = 7200;
+    const dstoff: i32 = 10800;
+    try testing.expectEqualSlices(u8, "IST", result.std_designation);
+    try testing.expectEqual(stdoff, result.std_offset);
+    try testing.expectEqualSlices(u8, "IDT", result.dst_designation.?);
+    try testing.expectEqual(dstoff, result.dst_offset);
+    // transition std 2020-03-27T01:59:59+02:00 --> dst 2020-03-27T03:00:00+03:00
+    try testing.expectEqual(stdoff, result.offset(1585267199).offset);
+    try testing.expectEqual(dstoff, result.offset(1585267200).offset);
+    // transtion dst 2020-10-25T01:59:59+03:00 --> std 2020-10-25T01:00:00+02:00
+    try testing.expectEqual(dstoff, result.offset(1603580399).offset);
+    try testing.expectEqual(stdoff, result.offset(1603580400).offset);
+}
+
+// FIXME : Buenos Aires has DST all year long
+test "posix TZ string, leap year, America/Argentina/Buenos_Aires" {
+    if (true) return error.SkipZigTest;
+    // IANA identifier: America/Argentina/Buenos_Aires
+    const result = try parsePosixTZ("WART4WARST,J1/0,J365/25");
+    const stdoff: i32 = -10800;
+    const dstoff: i32 = -10800;
+    try testing.expectEqualSlices(u8, "WART", result.std_designation);
+    try testing.expectEqualSlices(u8, "WARST", result.dst_designation.?);
+    try testing.expectEqual(stdoff, result.std_offset); // FIXME : this does not work yet
+    try testing.expectEqual(dstoff, result.dst_offset); // FIXME : this does not work yet
```

Comment:
This one is from the GNU/libc page. I think it is conceptually challenging since the standard offset is never active while in the code, only the DST offset is an optional. However, I'd consider this to be only an internal problem - to the "outside", the rule should return the DST offset and designation WARST, no matter what Unix time you give it.